rfcs/pagination-component.md

```diff
@@ -0,0 +1,67 @@
+# Pagination Component
+
+### Exports
+
+- `Pagination`
+- `StatefulPagination`
+- `StatefulContainer`
+- `StyledRoot`
+- `StyledBackButton`
+- `StyledNextButton`
+- `StyledMaxLabel`
+- `STATE_CHANGE_TYPE`
+
+### Pagination API
+
+- `numPages: number` - Required
+  Max number of pages
+- `currentPage: number` - Required
+  The current page
+- `prepositionLabel: string` - Optional
+  Defaults to "of"
+- `overrides: {PrevButton, NextButton, Select}` - Optional
+  Presentational override props, all are `React.ComponentType<*> | {component, props, style}`
+- `onBackClick: (event) => void` - Optional
+  Callback for prev button click
+- `onNextClick: (event) => void` - Optional
+  Callback for next button click
+
+### StatefulContainer API
+
+- `numPages: number` - Required
+  Max number of pages
+- `stateReducer: (changeType, changes, currentState) => Object`
+  State reducer function
+- `initialState: {currentPage: number}` - Optional
+  The initial state
+- `onPageChange: (nextPage, prevPage) => void` - Optional
+  Callback for when page changes
+
+### StatefulPagination API
+
+`StatefulContainer` API and the following
+
+- `prepositionLabel: string` - Optional
+  Defaults to "of"
+- `overrides: {PrevButton, NextButton, Select}` - Optional
+  Presentational override props, all are `React.ComponentType<*> | {component, props, style}`
+- `onBackClick: (event) => void` - Optional
+  Callback for prev button click
+- `onNextClick: (event) => void` - Optional
+  Callback for next button click
```

Comment:
Should StatefulPagination have `onPageChange` also? (Instead of, or in addition to onBackClick/onNextClick)

Reply:
Yep, it has everything that `StatefulContainer` has